manifests/freshclam.pp

```diff
@@ -25,6 +26,17 @@
     content => template("${module_name}/clamav.conf.erb"),
   }
 
+  if ($::osfamily == 'RedHat') and (versioncmp($::operatingsystemmajrelease, '7') >= 0) {
+    file { 'freshclam_sysconfig':
```

Comment:
How about changing the condition to just:
```puppet
if $clamav::freshclam_sysconfig {
}
```

Since you already check the OS and set that in params...